runtime/autoload/man.vim

```diff
@@ -1,137 +1,211 @@
-let s:man_tag_depth = 0
-let s:man_sect_arg = ''
-let s:man_find_arg = '-w'
-
-try
-  if !has('win32') && $OSTYPE !~? 'cygwin\|linux' && system('uname -s') =~? 'SunOS' && system('uname -r') =~? '^5'
-    let s:man_sect_arg = '-s'
-    let s:man_find_arg = '-l'
-  endif
-catch /E145:/
-  " Ignore the error in restricted mode
-endtry
-
-" Load man page {page} from {section}
-"   call man#get_page([{section}, ]{page})
-function man#get_page(...) abort
-  let invoked_from_man = (&filetype ==# 'man')
-
-  if a:0 == 0
-    echoerr 'argument required'
-    return
-  elseif a:0 > 2
-    echoerr 'too many arguments'
+let s:man_cmd = 'man 2>/dev/null'
+" regex for valid extensions that manpages can have
+let s:man_extensions = '[glx]z\|bz2\|lzma\|Z'
+let s:man_tag_stack = []
+function! man#get_page(bang, editcmd, ...) abort
+  " fpage is a string like 'printf(2)'
+  if empty(a:000)
+    let fpage = expand('<cWORD>')
+    if empty(fpage)
+      call s:error("no WORD under cursor")
+      return
+    endif
+  elseif len(a:000) > 2
+    call s:error('too many arguments')
     return
+  elseif len(a:000) == 1
+    let fpage = a:000[0]
+  else
+    let fpage = a:000[1].'('.a:000[0].')'
   endif
```

Comment:
This block is harder to follow than what you replaced. Why the change from `a:0` to `len(a:000)`? Why not rely on `get()`? Restoring and updating the documentation of the function parameters above would be good.